### PR TITLE
Fix out-of-bounds access

### PR DESCRIPTION
**Pull request by jlblancoc**
This was a bug, right?  :-)

We cannot write into `Vector3` [3].

### CODE REVIEW

File: pointmatcher/DataPointsFilters/CovarianceSampling.cpp [hunk 1]
```diff
@@ -104,7 +104,7 @@ void CovarianceSamplingDataPointsFilter<T>::inPlaceFilter(DataPoints& cloud)
 	for(std::size_t i = 0; i < featDim - 1; ++i) center(i) = T(0.);
 	
 	for (std::size_t i = 0; i < nbCandidates; ++i)
-		for (std::size_t f = 0; f <= 3; ++f)
+		for (std::size_t f = 0; f < 3; ++f)
 			center(f) += cloud.features(f,candidates[i]);
 	
 	for(std::size_t i = 0; i <= 3; ++i) center(i) /= T(nbCandidates);
```
Comment (pomerlef) [inline pointmatcher/DataPointsFilters/CovarianceSampling.cpp:110]: I suspect the same here

File: pointmatcher/DataPointsFilters/CovarianceSampling.cpp [hunk 2]
```diff
@@ -104,7 +104,7 @@ void CovarianceSamplingDataPointsFilter<T>::inPlaceFilter(DataPoints& cloud)
 	for(std::size_t i = 0; i < featDim - 1; ++i) center(i) = T(0.);
 	
 	for (std::size_t i = 0; i < nbCandidates; ++i)
-		for (std::size_t f = 0; f <= 3; ++f)
+		for (std::size_t f = 0; f < 3; ++f)
```
Comment (pomerlef) [inline pointmatcher/DataPointsFilters/CovarianceSampling.cpp:107]: good catch

**Comment by pomerlef** [comment]
ok to test

**Comment by YoshuaNava** [comment]
@jlblancoc Good eye :eye: 

Did you use sampled covariances in your work? How was your experience with them?

File: pointmatcher/DataPointsFilters/CovarianceSampling.cpp (hunk 1 above)
Comment (jlblancoc) [reply pointmatcher/DataPointsFilters/CovarianceSampling.cpp]: Oh, sure! Fixed too.

**Comment by jlblancoc** [comment]
>  Good eye eye

Merits should actually go to g++10 and properly-set flags ;-)

> Did you use sampled covariances in your work? How was your experience with them?

Actually not, sorry... RandomSamplingDataPointsFilter and SurfaceNormalDataPointsFilter worked really well, so didn't tested the other methods (at least, so far).

**Comment by pomerlef** [comment]
We've test it over here: https://arxiv.org/pdf/1810.01666.pdf if you're interested. Hopefully the bug didn't inluenced too much the results.

**Pull request merged**